packages/ast-position/lib/ast-position.js

```diff
@@ -0,0 +1,41 @@
+class AstPositionVisitor {
+  constructor(offset) {
+    this.offset = offset;
+  }
+  visitXMLElement(node) {
+    const openName = node.syntax.openName;
+    const closeName = node.syntax.closeName;
+    if (this.isOffsetInRange(openName)) {
+      this.astContext = { kind: "XMLElementOpenName", astNode: node };
+    }
+
+    if (this.isOffsetInRange(closeName)) {
+      this.astContext = { kind: "XMLElementCloseName", astNode: node };
+    }
+  }
+
+  visitXMLAttribute(node) {
+    const key = node.syntax.key;
+    const value = node.syntax.value;
+
+    if (this.isOffsetInRange(key)) {
+      this.astContext = { kind: "XMLKeyAttribute", astNode: node };
+    }
+
+    if (this.isOffsetInRange(value)) {
```

Comment:
Use `else if` here (if you already found the range in the attribute key you shouldn't check the attribute value too)